syntax = "proto3";

// The BackEnd Service definition.
service BackEndService {
  // Send an Inference Request
  rpc infer (TPURequest) returns (TPUResponse) {}
}

// The request message contains the image name, type and the image itself.
message TPURequest {
    string image_name = 1;
    string image_type = 2;
    bytes image = 3;            // Input image
    string model_name = 4;      // Name of the model
    string type = 5;            // Type of this request, now support 'classification' and 'detection'
}

// The response message containing the class
message TPUResponse {
  repeated bytes output = 1;
  double start_time = 2;
  double end_time = 3;
  double tpu_time = 4;
}
